### ISSUE TITLE
Action is checking out main branch and not respecting tags

### ISSUE THREAD

**Issue by Aaron-Ritter** [FusionAuth/fusionauth-github-action#3]
The way the action is build with the checkout action it is always checking out the main branch instead of using the github action directory directly.

https://github.com/FusionAuth/fusionauth-github-action/blob/85b1ff5fb9defddb5b2bdd0506cc478a6c359b80/action.yaml#L40-L44

This is especially a problem for consistent testing workflows. If you deliberately choose fusionauth/fusionauth-github-action@main version it is the only time where this is valid, but if you chose fusionauth/fusionauth-github-action@v1 or more precise fusionauth/fusionauth-github-action@v1.0.4 action you want to make sure to stay on these specific release tags.

**Comment by mooreds** [comment]
Is the answer to have another environment variable for this action that defaults to main but takes a tag?

**Comment by Aaron-Ritter** [comment]
No, much simpler, removing checkout all together and just using `github.action_path` will make sure to use the action in the version as defined in the workflow.

The PR contains that suggestion in more detail.
